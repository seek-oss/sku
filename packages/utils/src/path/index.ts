export { toPosixPath } from './toPosixPath.js';
