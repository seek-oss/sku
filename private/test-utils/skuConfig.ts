/**
 * A helper to make stable hashes for webpack.
 */
export const makeStableHashes = (config: Record<string, any>) => {
  // Addresses an issue where module IDs were different between local dev and CI
  config.optimization.chunkIds = 'natural';
  config.optimization.moduleIds = 'natural';

  // Disable hashes
  if (config.output.filename?.includes('[contenthash]')) {
    config.output.filename = '[name].js';
  }
  if (config.output.chunkFilename?.includes('[contenthash]')) {
    config.output.chunkFilename = '[name].chunk.js';
  }

  return config;
};

/**
 * A helper for vite to make stable hashes.
 */
export const makeStableViteHashes = () => ({
  build: {
    rollupOptions: {
      output: {
        chunkFileNames: '[name].js',
        assetFileNames: '[name][extname]',
        entryFileNames: '[name].js',
      },
    },
  },
});
