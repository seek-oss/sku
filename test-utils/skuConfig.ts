export const makeStableHashes = (config: Record<string, any>) => {
  // Addresses an issue where module IDs were different between local dev and CI
  config.optimization.chunkIds = 'natural';
  config.optimization.moduleIds = 'natural';

  // Disable hashes
  if (config.output.filename?.includes('[contenthash]')) {
    config.output.filename = '[name].js';
  }
  if (config.output.chunkFilename?.includes('[contenthash]')) {
    config.output.chunkFilename = '[name].chunk.js';
  }

  return config;
};
