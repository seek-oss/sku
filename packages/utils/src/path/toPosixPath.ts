import path from 'node:path';

/**
 * Replaces all win32 path separators (\) with posix path separators (/)
 */
export const toPosixPath = (inputPath: string) =>
  inputPath.replaceAll(path.win32.sep, path.posix.sep);
