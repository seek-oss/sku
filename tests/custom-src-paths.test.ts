import path from 'node:path';
import {
  dirContentsToObject,
  runSkuScriptInDir,
  waitForUrls,
  getAppSnapshot,
} from '@sku-private/test-utils';

import skuConfig from '@sku-fixtures/custom-src-paths/sku.config';
import type { ChildProcess } from 'node:child_process';
const appDir = path.dirname(
  require.resolve('@sku-fixtures/custom-src-paths/sku.config.ts'),
);

const targetDirectory = `${appDir}/dist`;
const url = `http://localhost:${skuConfig.port}`;

describe('custom-src-paths', () => {
  describe('start', () => {
    let process: ChildProcess;

    beforeAll(async () => {
      process = await runSkuScriptInDir('start', appDir);
      await waitForUrls(url);
    });

    afterAll(async () => {
      await process.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(url);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('build', () => {
    let process: ChildProcess;

    beforeAll(async () => {
      await runSkuScriptInDir('build', appDir);
      process = await runSkuScriptInDir('serve', appDir);
      await waitForUrls(url);
    });

    afterAll(async () => {
      await process.kill();
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(targetDirectory);
      expect(files).toMatchSnapshot();
    });

    it('should create valid app', async () => {
      const app = await getAppSnapshot(url);
      expect(app).toMatchSnapshot();
    });
  });

  describe('format', () => {
    it('should format successfully', async () => {
      const { child } = await runSkuScriptInDir('format', appDir);
      expect(child.exitCode).toEqual(0);
    });
  });

  describe('lint', () => {
    it('should lint successfully', async () => {
      const { child } = await runSkuScriptInDir('lint', appDir);
      expect(child.exitCode).toEqual(0);
    });
  });
});
