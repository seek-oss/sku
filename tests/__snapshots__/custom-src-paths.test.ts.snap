// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`custom-src-paths > bundler vite > build > should create valid app 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      rel="modulepreload"
      href="/some-static-place/vite-client.js"
      crossorigin
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello custom src paths
      </div>
    </div>
    <script
      type="module"
      src="/some-static-place/vite-client.js"
    >
    </script>
  </body>
</html>"
POST HYDRATE DIFFS: NO DIFF
`;

exports[`custom-src-paths > bundler vite > build > should generate the expected files 1`] = `
{
  "index.html": "<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>hello-world</title>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="modulepreload" href="/some-static-place/vite-client.js" crossorigin />

  </head>
  <body>
    <div id="app"><div>Hello custom src paths</div></div>
    <script type="module" src="/some-static-place/vite-client.js" ></script>
  </body>
</html>",
  "vite-client.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`custom-src-paths > bundler vite > start > should start a development server 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <script type="module">
      import { injectIntoGlobalHook } from "/@react-refresh"
injectIntoGlobalHook(window);
window.$RefreshReg$ = () => {};
window.$RefreshSig$ = () => (type) => type;
    </script>
    <script
      type="module"
      src="/@vite/client"
    >
    </script>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <script
      type="module"
      src="/@id/__x00__/index.html?html-proxy&index=0.js"
    >
    </script>
    <script
      type="module"
      src="/@id/__x00__/index.html?html-proxy&index=1.js"
    >
    </script>
  </head>
  <body>
    <div id="app">
      <div>
        Hello custom src paths
      </div>
    </div>
    <script
      type="module"
      src="{cwd}/packages/sku/dist/services/vite/entries/vite-client.js"
    >
    </script>
  </body>
</html>"
POST HYDRATE DIFFS: 
===================================================================
--- sourceHtml
+++ clientHtml
@@ -22,12 +22,12 @@
     >
     <script
       type="module"
-      src="/@id/__x00__/index.html?html-proxy&index=0.js"
+      src="/@id/__x00__/index.html?html-proxy&amp;index=0.js"
     >
     </script>
     <script
       type="module"
-      src="/@id/__x00__/index.html?html-proxy&index=1.js"
+      src="/@id/__x00__/index.html?html-proxy&amp;index=1.js"
     >
     </script>
   </head>
`;

exports[`custom-src-paths > bundler webpack > build > should create valid app 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="/some-static-place/runtime.js"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="/some-static-place/2.js"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="/some-static-place/main.js"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello custom src paths
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      []
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script
      async
      data-chunk="main"
      src="/some-static-place/runtime.js"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="/some-static-place/2.js"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="/some-static-place/main.js"
    >
    </script>
  </body>
</html>"
POST HYDRATE DIFFS: NO DIFF
`;

exports[`custom-src-paths > bundler webpack > build > should generate the expected files 1`] = `
{
  "2.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "2.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "index.html": "<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>hello-world</title>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    
<link data-chunk="main" rel="preload" as="script" href="/some-static-place/runtime.js">
<link data-chunk="main" rel="preload" as="script" href="/some-static-place/2.js">
<link data-chunk="main" rel="preload" as="script" href="/some-static-place/main.js">
  </head>
  <body>
    <div id="app"><div>Hello custom src paths</div></div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__" type="application/json">[]</script><script id="__LOADABLE_REQUIRED_CHUNKS___ext" type="application/json">{"namedChunks":[]}</script>
<script async data-chunk="main" src="/some-static-place/runtime.js"></script>
<script async data-chunk="main" src="/some-static-place/2.js"></script>
<script async data-chunk="main" src="/some-static-place/main.js"></script>
  </body>
</html>",
  "main.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`custom-src-paths > bundler webpack > start > should start a development server 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="/runtime.js"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="/2.js"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="/main.js"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello custom src paths
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      []
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script
      async
      data-chunk="main"
      src="/runtime.js"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="/2.js"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="/main.js"
    >
    </script>
  </body>
</html>"
POST HYDRATE DIFFS: NO DIFF
`;
